src/attestation/azext_attestation/generated/_validators.py

```diff
@@ -7,3 +7,19 @@
 # Changes may cause incorrect behavior and will be lost if the code is
 # regenerated.
 # --------------------------------------------------------------------------
+
+from knack.util import CLIError
+
+from ._client_factory import cf_attestation_provider
+
+
+def validate_attestation_name(cmd, ns):
+    if not ns.tenant_base_url:
+        if ns.attestation_name and ns.resource_group_name:
+            provider = cf_attestation_provider(cmd.cli_ctx).\
+                get(provider_name=ns.attestation_name, resource_group_name=ns.resource_group_name)
+            ns.tenant_base_url = provider.attest_uri
+            del ns.attestation_name
+            del ns.resource_group_name
+        else:
+            raise CLIError('incorrect usage: [--attestation-base-url | --name/-n --resource-group-name/-g]')
```

Comment:
It'll not be allowed to do without providing "tenant_base_url and attestation_name/rg_name" or all three params. For example, `az attestation policy set --tee SgxEnclave -p "myprovider"` would not work even though all required arguments are given.

Reply:
Hi @YalinLi0312 , this is a trade-off in CLI. In some scenarios, you need to choose between param A and param B to specify (if you specify A you can ignore B, vice versa). Both A and B are not `required`, and we use validators to check the rule. If we mark A as `required`, the framework would always require it to be specified, even B is provided.
This is a similar case: https://github.com/Azure/azure-cli/issues/14630